eng/Generate.ps1

```diff
@@ -1,64 +1,58 @@
+    {
+        throw "Command failed to execute: $command"
     }
 }
 
-function Invoke-AutoRest($autoRestArguments, $repoRoot) {
-    Invoke-Block {
-        $command = "npx autorest-beta $autoRestArguments"
-        $commandText = $command.Replace($repoRoot, "`$(SolutionDir)")
-
-        Write-Host ">" $commandText
-        
-        & cmd /c "$command 2>&1"
+function Invoke-Generate($name, [switch]$noDebug)
+{
+    # General configuration
+    $repoRoot = Resolve-Path (Join-Path $PSScriptRoot '..')
+    $debugFlags = if (-not $noDebug) { '--debug', '--verbose' }
+
+    # Test server test configuration
+    $testServerDirectory = Join-Path $repoRoot 'test' 'TestServerProjects'
+    $configurationPath = Join-Path $testServerDirectory 'readme.tests.md'
+    $testServerSwaggerPath = Join-Path $repoRoot 'node_modules' '@microsoft.azure' 'autorest.testserver' 'swagger'
+    $testNames = if ($name) { $name } else { 'url', 'body-string', 'body-complex', 'custom-baseUrl', 'custom-baseUrl-more-options', 'header' }
+
+    foreach ($testName in $testNames)
+    {
+        $inputFile = Join-Path $testServerSwaggerPath "$testName.json"
+        $namespace = $testName.Replace('-', '_')
+        $autoRestArguments = "$debugFlags --require=$configurationPath --input-file=$inputFile --title=$testName --namespace=$namespace"
+        Invoke-AutoRest $autoRestArguments
     }
-}
-
-# General configuration
-$repoRoot = Resolve-Path (Join-Path $PSScriptRoot '..')
-$debugFlags = if (-not $noDebug) { '--debug', '--verbose' }
-
-# Test server test configuration
-$testServerDirectory = Join-Path $repoRoot 'test' 'TestServerProjects'
-$configurationPath = Join-Path $testServerDirectory 'readme.tests.md'
-$testServerSwaggerPath = Join-Path $repoRoot 'node_modules' '@microsoft.azure' 'autorest.testserver' 'swagger'
-$testNames = if ($name) { $name } else { 'url', 'body-string', 'body-complex', 'custom-baseUrl', 'custom-baseUrl-more-options', 'header' }
 
-if (!$NoReset)
-{
-    Invoke-AutoRest "--reset" $repoRoot
-}
+    # Sample configuration
+    $sampleDirectory = Join-Path $repoRoot 'samples'
+    $projectNames = 'AppConfiguration'
 
-foreach ($testName in $testNames)
-{
-    $inputFile = Join-Path $testServerSwaggerPath "$testName.json"
-    $namespace = $testName.Replace('-', '_')
-    $autoRestArguments = "$debugFlags --require=$configurationPath --input-file=$inputFile --title=$testName --namespace=$namespace"
-    Invoke-AutoRest $autoRestArguments $repoRoot
+    foreach ($projectName in $projectNames)
+    {
+        $projectDirectory = Join-Path $sampleDirectory $projectName
+        $configurationPath = Join-Path $projectDirectory 'readme.md'
+        $autoRestArguments = "$debugFlags --require=$configurationPath"
+        Invoke-AutoRest $autoRestArguments
+    }
 }
 
-# Sample configuration
-$sampleDirectory = Join-Path $repoRoot 'samples'
-$projectNames = 'AppConfiguration'
-
-foreach ($projectName in $projectNames)
+if(-not $noRun)
```

Comment:
This is a very strange pattern.

Reply:
When the 'executable' is also a 'library', it made it difficult to source and execute. Will fix the pattern in the morning.